### DISCUSSION TITLE
SQLE 如何升级版本

### DISCUSSION THREAD

**Discussion by sjjian** [actiontech/sqle#433] [Q&A] [answered]
RT

**Comment by sjjian** [comment, marked as answer]
## 升级方案
有些版本会有不兼容的变更，需要进行改表操作，参考：https://github.com/actiontech/sqle/discussions/categories/upgrade 。 **升级库表前先停止SQLE进程，升级库表完成后可按下面的步骤rpm升级下程序**
## RPM 安装的SQLE
```bash
rpm -Uvh --force {your rpm file} --prefix={your install file}
```
例如：
```bash
rpm -Uvh --force sqle-ce-1.2203.0.qa.el7.x86_64.rpm --prefix=/opt/sqle
```

**Comment by HuangWeiCen** [comment]
# docker安装的SQLE
1. 将sqle rpm包拷贝到容器内
`docker cp {rpm包} {容器ID}:/`
2. 使用root用户登入容器
`docker exec -it -uroot {容器ID}`
3. 和正常升级RPM包安装的SQLE一样升级
https://github.com/actiontech/sqle/discussions/433
4. 退出容器后重新启动容器
`docker restart {容器ID}`
